application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksSystem.java

```diff
@@ -0,0 +1,185 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.channel.ChannelType;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion;
+import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.BookmarksRecord;
+
+import javax.annotation.Nullable;
+
+import java.awt.*;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import static org.togetherjava.tjbot.db.generated.tables.Bookmarks.BOOKMARKS;
+
+/**
+ * The bookmarks system provides methods to interact with the database. It also enables the
```

Comment:
this description is kinda missing the core of what the system is for:

its the class that maintains the bookmarks.

Reply:
I dont fully understand what you mean. Could you be more specific or provide an example?

Reply:
> Maintains all bookmarks for all users. Provides means to create, get and delete them.
>
> Main user is {@link BookmarksCommand}.

Reply:
(still open)

Reply:
Yeah I know. I have something but did not commit yet

Reply:
How do you like
```
/**
 * Maintains all bookmarks for all users and provides methods to create, query and remove them.
 * Used by the other bookmarks classes.
 */
```

Reply:
sure, works

Reply:
Good. I won't be home until Saturday so feel free to change it and resolve the conflicts if you want to